resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/BasicDriverFeatureStep.scala

```diff
@@ -129,6 +132,7 @@ private[spark] class BasicDriverFeatureStep(conf: KubernetesDriverConf)
         .addToLimits(maybeCpuLimitQuantity.toMap.asJava)
         .addToRequests("memory", driverMemoryQuantity)
         .addToLimits("memory", driverMemoryQuantity)
+        .addToLimits(driverResourceQuantities.toMap.asJava)
```

Comment:
Why only apply them to the limit?

Reply:
I'm going by https://kubernetes.io/docs/tasks/manage-gpus/scheduling-gpus/
-> GPUs are only supposed to be specified in the limits 

also assuming that all resource plugins use limits going by: https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/
where it just has:
 containers:
    - name: demo-container-1
      image: k8s.gcr.io/pause:2.0
      resources:
        limits:
          vendor-domain/resource: 2 # requesting 2 vendor-domain/resource

If there is other documentation that says otherwise  please let me know

Reply:
You are right. The request will be set by default to the limit value.